use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,153 @@
+
+```bash
+mkdir -p /var/run/kata-containers/vhost-user/
+mkdir -p /var/run/kata-containers/vhost-user/block/
+mkdir -p /var/run/kata-containers/vhost-user/block/sockets/
+mkdir -p /var/run/kata-containers/vhost-user/block/devices/
+```
+
+Next, start the SPDK vhost target application.  The following command will start
+vhost on CPU cores 0 with all future socket files placed in `/var/run/kata-containers/vhost-user/block/sockets/`.
+
+```bash
+$ sudo app/spdk_tgt/spdk_tgt -S /var/run/kata-containers/vhost-user/block/sockets/ &
+```
+
+To list all available vhost options use the following command.
+
+```bash
+$ app/spdk_tgt/spdk_tgt -h
+```
+
+Create a experimental `vhost-user-blk` device based on memory directly.
+
+- The following RPC will create a 64MB memory block device with 4096-byte block
+size.
+
+```bash
+$ sudo scripts/rpc.py bdev_malloc_create 64 4096 -b Malloc0
+```
+
+- The following RPC will create a `vhost-user-blk` device exposing Malloc0 block
+device. The device will be accessible via
+`/var/run/kata-containers/vhost-user/block/sockets/vhostblk0`.
+
+```bash
+$ sudo scripts/rpc.py vhost_create_blk_controller vhostblk0 Malloc0
+```
+
+## Host setup for vhost-user devices
+
+Considering OCI spec and characteristic of vhost-user device, Linux reserved
+block major range (240-254) to map each vhost-user block type to a major。
+Also a specific directory is used for vhost-user devices. The base directory is
+a configurable value, with the default being `/var/run/kata-containers/vhost-user`.
+It can be configured by parameter `vhost_user_store_path` in Kata TOML
+configuration file.
+
+The reset of the path `block` is used for block device; `block/sockets` is where
+we expect vhost-user sockets to live; `block/sockets` is where simulated block
+device node for vhost-user devices to live.
+
+For example, if using the default directory `/var/run/kata-containers/vhost-user`,
+sockets for vhost-user device is under
+
+```
+/var/run/kata-containers/vhost-user/block/sockets/
+```
+
+Device node for vhost-user device is under
+
+```
+/var/run/kata-containers/vhost-user/block/devices/
+```
+
+Currently, block major number 241 maps to `vhost-user-blk` device type. For
+`vhost-user-blk` device `vhostblk0`, create a block device node with major 241
+and minor 0 for it in order to be recognized by Kata runtime.
+
+```bash
+$ sudo mknod /var/run/kata-containers/vhost-user/block/devices/vhostblk0 b 241 0
+```
+
+## Launch a Kata container with SPDK vhost-user block device
+
+To use `vhost-user-blk` device, use Docker to add a host `vhost-user-blk` device
+to the container. In docker, `--device=HOST-DIR:CONTAINER-DIR` is used to add
+a host device to the container.
+
+For example:
+
+```bash
+$ sudo docker run --runtime kata-runtime --device=/var/run/kata-containers/vhost-user/block/devices/vhostblk0:/dev/vda -it busybox sh
+```
+
+Do IO operation on the `vhost-user-blk` device inside container.
+
+```
+/ # ls -l /dev/vda
```

Comment:
what means `/` before `#` ? can we use `$ sudo ...` ?

Reply:
After docker run a container, "/ #" is shown instead of "$". So here "/ #" is used.

Reply:
I would drop the leading `/` here.